src/main/java/seedu/address/model/person/Person.java

```diff
@@ -75,6 +76,36 @@ public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
+    /**
+     * Add grade to gradeList
+     *
+     * @param grade New grade to be added
+     * @return new immutable Person
+     */
+    public Person addGrade(Grade grade) {
+        requireAllNonNull(grade);
+
+        GradeList newGradeList = this.gradeList.addGrade(grade);
+
+        return new Person(this.name, this.phone, this.email, this.address, this.tags, newGradeList,
+                          this.attendanceList);
+    }
+
+    /**
+     * Remove grade from gradeList based on index.
+     *
+     * @param index The index to remove the grade.
+     * @return new immutable Person
+     */
+    public Person removeGrade(Index index) {
+        requireAllNonNull(index);
+
+        GradeList newGradelist = this.gradeList.removeGrade(index);
+
+        return new Person(this.name, this.phone, this.email, this.address, this.tags, newGradelist,
+                          this.attendanceList);
```

Comment:
Another 16 spaces

Reply:
Because it is a continuation of the constructor, so it should align with it.